src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala

```diff
@@ -86,8 +90,9 @@ class CassandraSnapshotStore(cfg: Config) extends SnapshotStore with CassandraSt
 
   override def loadAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Option[SelectedSnapshot]] = {
     for {
-      prepStmt <- preparedSelectSnapshotMetadataForLoad
-      mds <- metadata(prepStmt, persistenceId, criteria).map(_.take(maxLoadAttempts))
+      prepStmt <- if (criteria.maxTimestamp == Long.MaxValue) preparedSelectSnapshotMetadataWithMaxLoadAttemptsLimit
+      else preparedSelectSnapshotMetadata
```

Comment:
I'm not sure I get this `Long.MaxValue` is used as a magic value to select a different select here?

Also: I'd like this better with a linefeed and both branches in the same intendentation level.